### ISSUE TITLE
Fine tuning training keeps saving every 2 epoch even though that's not how I set it up

### ISSUE THREAD

**Issue by GUUser91** [Plachtaa/seed-vc#113]
This is how I set it up

```
python train.py --config /mnt/exhdd/seed-vc/configs/presets/config_dit_mel_seed_uvit_whisper_small_wavenet.yml --dataset-dir /mnt/exhdd/seed-vc/Datasets/San-Diego/ --run-name San-Diego --batch-size 8 --max-steps 1000 --max-epochs 1000 --save-every 500 --num-workers 0
```

But it keeps saving every 2 epochs

```
/mnt/exhdd/seed-vc/venv/lib/python3.11/site-packages/torch/nn/utils/weight_norm.py:30: UserWarning: torch.nn.utils.weight_norm is deprecated in favor of torch.nn.utils.parametrizations.weight_norm.
  warnings.warn("torch.nn.utils.weight_norm is deprecated in favor of torch.nn.utils.parametrizations.weight_norm.")
Loaded checkpoint './checkpoints/models--myshell-ai--OpenVoiceV2/snapshots/f36e7edfe1684461a8343844af60babc2efbb727/converter/checkpoint.pth'
missing/unexpected keys: [] []
/mnt/exhdd/seed-vc/venv/lib/python3.11/site-packages/torch/nn/utils/weight_norm.py:30: UserWarning: torch.nn.utils.weight_norm is deprecated in favor of torch.nn.utils.parametrizations.weight_norm.
  warnings.warn("torch.nn.utils.weight_norm is deprecated in favor of torch.nn.utils.parametrizations.weight_norm.")
Loading weights from nvidia/bigvgan_v2_22khz_80band_256x
Removing weight norm...
Warning: Skipped loading some keys due to shape mismatch: {'estimator.f0_embedder.weight', 'estimator.input_pos'}
cfm loaded
length_regulator loaded
Loaded checkpoint from ./checkpoints/models--Plachta--Seed-VC/snapshots/d9d1850f97c1820633956d7e4763ccd908ff9c1d/DiT_seed_v2_uvit_whisper_small_wavenet_bigvgan_pruned.pth
  0%|                                                                                                                                                             | 0/2 [00:00<?, ?it/s]max value is  tensor(1.0068)
max value is  tensor(1.0068)
max value is  tensor(1.0068)
max value is  tensor(1.0068)
/mnt/exhdd/seed-vc/venv/lib/python3.11/site-packages/torch/functional.py:650: UserWarning: stft with return_complex=False is deprecated. In a future pytorch release, stft will retur
max value is  tensor(1.0068)
max value is  tensor(1.0068)
max value is  tensor(1.0068)
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:01<00:00,  1.37it/s]
Epoch completed. Saving..
Checkpoint saved at ./runs/San-Diego/DiT_epoch_00033_step_00068.pth
  0%|                                                                                                                                                             | 0/2 [00:00<?, ?it/s]max value is  tensor(1.0068)
max value is  tensor(1.0068)
max value is  tensor(1.0068)
 50%|██████████████████████████████████████████████████████████████████████████▌                                                                          | 1/2 [00:01<00:01,  1.02s/it]max value is  tensor(1.0068)
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:01<00:00,  1.30it/s]
Epoch completed. Saving..
Checkpoint saved at ./runs/San-Diego/DiT_epoch_00034_step_00070.pth
  0%|                                                                                                                                                             | 0/2 [00:00<?, ?it/s]max value is  tensor(1.0068)
max value is  tensor(1.0068)
max value is  tensor(1.0068)
max value is  tensor(1.0068)
epoch 35, step 70, loss: 0.6484248942939579
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:01<00:00,  1.34it/s]
Epoch completed. Saving..
Checkpoint saved at ./runs/San-Diego/DiT_epoch_00035_step_00072.pth
  0%|                                                                                                                                                             | 0/2 [00:00<?, ?it/s]max value is  tensor(1.0068)
max value is  tensor(1.0068)
 50%|██████████████████████████████████████████████████████████████████████████▌                                                                          | 1/2 [00:01<00:01,  1.01s/it]max value is  tensor(1.0068)
max value is  tensor(1.0068)
100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:01<00:00,  1.32it/s]
Epoch completed. Saving..
```

**Comment by Hugo-san** [comment]
Actually, from your log, it means that the model saved the checkpoint every epochs and this is how the train.py is designed. 
`def train(self):
        self.ema_loss = 0
        self.loss_smoothing_rate = 0.99
        for epoch in range(self.n_epochs):
            self.epoch = epoch
            self.train_one_epoch()
            # Save after each epoch
            print('Epoch completed. Saving..')
            state = {
                'net': {key: self.model[key].state_dict() for key in self.model},
                'optimizer': self.optimizer.state_dict(),
                'scheduler': self.optimizer.scheduler_state_dict(),
                'iters': self.iters,
                'epoch': self.epoch,
            }
            save_path = os.path.join(
                self.log_dir,
                f'DiT_epoch_{self.epoch:05d}_step_{self.iters:05d}.pth'
            )
            torch.save(state, save_path)
            print(f"Checkpoint saved at {save_path}")

            if self.iters >= self.max_steps:
                break`

The parameter "--save-every 500" set the model to save every 500 steps instead of epochs. Your "--max-steps" is 1000 so it will only save twice for each 500 steps. The point here is just to differentiate the "epochs" and "steps" and you will understand it.

**Comment by GUUser91** [comment]
@Hugo-san
How do I get it to stop saving every epoch. I used an old train.py file from a fork to stop saving evey 1 epoch.

**Comment by Hugo-san** [comment]
> @Hugo-san How do I get it to stop saving every epoch. I used an old train.py file from a fork to stop saving evey 1 epoch.

Maybe you can simply delete this part of the code from above to make it stop saving every epoch:

# Save after each epoch
print('Epoch completed. Saving..')
state = {
'net': {key: self.model[key].state_dict() for key in self.model},
'optimizer': self.optimizer.state_dict(),
'scheduler': self.optimizer.scheduler_state_dict(),
'iters': self.iters,
'epoch': self.epoch,
}
save_path = os.path.join(
self.log_dir,
f'DiT_epoch_{self.epoch:05d}step{self.iters:05d}.pth'
)
torch.save(state, save_path)
print(f"Checkpoint saved at {save_path}")

**Comment by GUUser91** [comment]
@Hugo-san
That did it. Thanks.
